src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useEffect, useMemo } from "react";
+import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useAsyncFn } from "react-use";
+import firebase from "firebase/app";
+
+import { MAX_TABLE_CAPACITY, MIN_TABLE_CAPACITY } from "settings";
+
+import { Table } from "types/Table";
+
+import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
+
+import { InputField } from "components/atoms/InputField";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableForm {
+  title: string;
+  subtitle?: string;
+  capacity: number;
+}
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  subtitle?: string;
+  capacity: number;
+  venueName: string;
+  tables: Table[];
+  tableOfUser?: Table;
+  onHide: () => void;
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  venueName,
+  title,
+  tables,
+  tableOfUser,
+  subtitle,
+  capacity,
+  onHide,
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+
+  const formDefaultValues = useMemo(
+    () => ({
+      title,
+      subtitle,
+      capacity,
+    }),
+    [capacity, subtitle, title]
+  );
+
+  const {
+    register,
+    handleSubmit,
+    errors,
+    watch,
+    reset,
+  } = useForm<EditTableForm>({
+    mode: "onBlur",
+    reValidateMode: "onBlur",
+    defaultValues: formDefaultValues,
+  });
```

Comment:
> Do the defaultValues passed to useForm actually need to be useMemo'd?

Yes, they are being used in a useEffect, and it's cry that the object is not memoed

Reply:
I'm also using the same object for the `reset` function. 

> In the 'rules' section for defaultValues it says that they are used for the first render:

Yes, but after the props change, these values don't. And If I open the modal and someone else updates the table info at that moment, my modal data will be outdated.

Reply:
They are memoed only because of the `useEffect` otherwise I get this message.

```tsx
The 'formDefaultValues' object makes the dependencies of useEffect Hook (at line 67) change on every render. To fix this, wrap the initialization of 'formDefaultValues' in its own useMemo() Hook.
```